File: commons/javafx/src/main/java/com/cedarsoft/commons/javafx/IntegerTextFormatterFilter.java
package com.cedarsoft.commons.javafx;

import java.util.function.Predicate;

import javax.annotation.Nonnull;

import javafx.util.converter.NumberStringConverter;

/**
 * Processes text formatter changes. Ensures only valid integers are added
 */
class IntegerTextFormatterFilter extends NumberTextFormatterFilter {

  IntegerTextFormatterFilter(@Nonnull NumberStringConverter converter, @Nonnull Predicate<Integer> filter) {
    super(converter, number -> filter.test(number.intValue()), 0);
  }

}

File: commons/javafx/src/main/java/com/cedarsoft/commons/javafx/IntegerTextFormatterFilter.kt
package com.cedarsoft.commons.javafx

import javafx.util.converter.NumberStringConverter
import java.util.function.Predicate

/**
 * Processes text formatter changes. Ensures only valid integers are added
 */
class IntegerTextFormatterFilter(
  converter: NumberStringConverter,
  filter: Predicate<Int>,
) : NumberTextFormatterFilter(
  converter = converter,
  filter = Predicate { number: Number -> filter.test(number.toInt()) },
  emptyValue = 0
)
